### PR TITLE
crimson/seastore: add extent replay support, implement for lba tree nodes

### PR DESCRIPTION
**Pull request by athanatos**
Also cleans up some details about how the root node is handled -- see the related commit.

### CODE REVIEW

**Comment by athanatos** [comment]
rebasing.

**Comment by athanatos** [comment]
@liu-chunmei You may also want to take a look -- I reworked FixedKVNodeLayout so that it provides replay mostly for free.

File: src/crimson/common/fixed_kv_node_layout.h [hunk 1]
```diff
@@ -126,113 +138,291 @@ class FixedKVNodeLayout {
...
-    memcpy(
-      tgt->get_val_ptr(), from_src->get_val_ptr(),
-      to_src->get_val_ptr() - from_src->get_val_ptr());
-    memcpy(
-      tgt->get_key_ptr(), from_src->get_key_ptr(),
-      to_src->get_key_ptr() - from_src->get_key_ptr());
-  }
+  bool operator==(const FixedKVNodeLayout &rhs) const {
+    if (get_size() != rhs.get_size()) {
+      assert(false);
```
Comment (tchaikov) [inline src/crimson/common/fixed_kv_node_layout.h:410]: do you mean, it's not expected that `lhs != rhs` ?

**Comment by tchaikov** [comment]
jenkins test docs

File: src/crimson/common/fixed_kv_node_layout.h (hunk 1 above)
Comment (athanatos) [reply src/crimson/common/fixed_kv_node_layout.h:410]: Mistake, left over from debugging, will remove, thanks.

File: src/crimson/os/seastore/CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@ add_library(crimson-seastore
   onode_manager/simple-fltree/onode_delta.cc
   onode_manager/simple-fltree/onode_node.cc
   seastore.cc
+  ../../../test/crimson/seastore/test_block.cc
```
Comment (tchaikov) [inline src/crimson/os/seastore/CMakeLists.txt:17]: why? i don't think it's a part of seastore.

Comment (athanatos) [reply src/crimson/os/seastore/CMakeLists.txt:17]: cache.cc needs to be able to generate a TestBlock object for unittests.  At some point I may make it more dynamic, but this is simpler for now.

**Pull request merged**